### ISSUE TITLE
[FR] Add some useful goto options - tomorrow & yesterday

### ISSUE THREAD

**Issue by TC72** [renerocksai/telekasten.nvim#296]
Add more goto options like `goto_today`. I now live in telekasten for everything I'm working on and it would be nice to have a simple way to go to tomorrow's journal and write myself notes to get that workday started on a roll. Being able to go back to yesterday to find all the todos I failed to complete would also be nice.
I can add this myself but wanted to raise it as an issue before submitting a PR. 

I'd thought about suggesting this before but I see that Obsidian.nvim goes a step further and goto tomorrow and yesterday can figure out which work day to go to. I don't need that feature myself but would be good to cover all bases.

**Comment by lambtho12** [comment]
I agree with the proposal. And I am happy that you are willing to make it happen! Thank you very much for getting involved! If you are motivated, feel free to tackle other stuff as well :smile: 

----
Now, let's be smart (and lazy) about that implementation.. Rather than defining a new function, we could just add an `offset` option to `GotoToday` (or ultimately `GotoDate`) that would let user define their mappings or functions as they want (yesterday `offset=-1`, same day last week `offset=-7`, tomorrow `offset=1`, same day next week `offset=7`, in three days `offset=3`, whatever).

E.g.
```lua
  vim.keymap.set(
    "n",
    "<leader>zY",
    ":lua require('telekasten').goto_today({offset=-1})<CR>",
    { desc = "Go to [Y]esterday" }
  )
```

If you want to really go over board, we could also have a parser so we could say (GotoDate("next sunday")) or something, but that is probably much complex.

**Comment by TC72** [comment]
This works and handles no offset being provided.

My only experience with Lua is writing a simple telescope extension so any feedback on whether this is the right way to do things would be great. I've just added an extra parameter into the `os.date()` call instead of it defaulting to using `os.time()` to get now.

```
local function GotoToday(opts)
    opts = opts or {}
    opts.offset = opts.offset or 0

    global_dir_check(function(dir_check)
        if not dir_check then
            return
        end

        local today = os.date(dateutils.dateformats.date,(os.time()+(86400*opts.offset)))
        opts.date_table = os.date("*t")
        opts.date = today
        opts.dailies_create_nonexisting = true -- Always use template for GotoToday
        GotoDate(opts)
    end)
end
```
